### ISSUE TITLE
Detect duplicate `case` clauses

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#11532]
## Is your feature request related to a problem? Please describe.

Detect duplicate clauses in a case statement that can be combined.

## Describe the solution you'd like

Warning of duplicate clauses that can be combined, such as:

```ruby
case 'foo'
when 1
  'bar'
when 2
  'bar'
when 3
  'baz'
end
```

Optionally, support auto-correction to:

```ruby
case 'foo'
when 1, 2
  'bar'
when 3
  'baz'
end
```

## Describe alternatives you've considered

I couldn't find any cop which does this currently.

## Additional context

This would closely related to [Lint/DuplicateCaseCondition](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/DuplicateCaseCondition)

**Comment by fatkodima** [comment]
This - https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/DuplicateBranch ?

**Comment by andyw8** [comment]
Yes! Thank you. It didn't come up in my search, and for some reason is [disabled](https://github.com/Shopify/ruby-style-guide/blob/f9e67645d69e95b171688f5ebcacf3696ca4c11a/rubocop.yml#L121-L122) in my config.
